include/glow/Importer/ONNX.h

```diff
@@ -48,28 +48,46 @@ class ONNXModelLoader
   /// Load the network initializers from the GraphProto.
   void loadInitializers(onnx::GraphProto &net);
 
+  /// \returns true if operator \p op can be loaded.
   /// Load the operator \p op into the network. This creates one or more nodes
   /// in the network.
-  void loadOperator(const onnx::NodeProto &op);
+  bool loadOperator(const onnx::NodeProto &op);
 
-  /// Reads a network (weights or structure) from the serialized protocol buffer
-  /// file.
-  bool loadProtoFile(onnx::GraphProto &net, const std::string &filename);
+  /// \returns true if \p net can be constructed from the content of the
+  /// file \p filename.
+  /// Loads GraphProto \p net from the file containing serialized protobuf.
+  bool loadProto(onnx::GraphProto &net, const std::string &filename);
+
+  /// \returns true if \p net can be constructed from the in-memory
+  /// serialized protobuf.
+  /// Loads GraphProto \p net from the in-memory serialized protobuf \p onnxModel
+  /// with the model size \p onnxModelSize.
+  bool loadProto(onnx::GraphProto &net, const void *onnxModel, size_t onnxModelSize);
+
+  /// \returns true if GraphProto \p net can be loaded from the stream \p iStream.
+  bool loadProto(onnx::GraphProto &net, google::protobuf::io::ZeroCopyInputStream& iStream);
+
+  /// Creates a ONNX model loader to build \p F.
+  ONNXModelLoader(Function &F);
 
   /// ONNX model ir_version;
   size_t irVersion_;
 
   /// ONNX model op_version;
   size_t opsetVersion_;
-
 public:
   /// Loads the ONNX model that's represented by a model description file,
   /// serialized in \p modelDescFilename and populates the network into \p F.
   /// The tensors in \p tensors are stored with the names in the list of names
-  /// \p names and used as inputs to the network.
+  /// \p tensorNames and used as inputs to the network.
   ONNXModelLoader(const std::string &modelDescFilename,
-                  llvm::ArrayRef<const char *> names,
+                  llvm::ArrayRef<const char *> tensorNames,
```

Comment:
Maybe it's worth to change names -> tensorNames in all 3 constructors: also in ProtobufLoader and Caffe2ModelLoader.

Reply:
Yup, will change in the follow-up PR.

Reply:
fixed it here: https://github.com/pytorch/glow/pull/1301